src/dotnet-core-uninstall/Windows/UninstallCommandExec.cs

```diff
@@ -48,6 +56,38 @@ public static void ExecuteUninstallCommand(IEnumerable<Bundle> bundles)
                     }
                 };
 
+                var cancelProcessHandler = new ConsoleCancelEventHandler((sender, cancelArgs) =>
+                {
+                    cancelMutex.WaitOne();
+
+                    if (!canceled)
+                    {
+                        canceled = true;
+                        Console.WriteLine(LocalizableStrings.CancelingMessage);
+                    }
+
+                    cancelArgs.Cancel = true;
+
+                    cancelMutex.ReleaseMutex();
+                });
+
+                Console.CancelKeyPress += cancelProcessHandler;
+
+                process.Exited += new EventHandler((sender, e) =>
```

Comment:
I think we can just do a check for `canceled` after the `WaitForExit` call rather than using the `Exited` event and an explicit `Environment.Exit` call.

Something like:

```c#
...

if (process.ExitCode != 0)
{
    throw new UninstallationFailedException(bundle.UninstallCommand, process.ExitCode);
}

cancelMutex.WaitOne();

try
{
    if (canceled)
    {
        // Either break here for a graceful exit or call `Environment.Exit(1)`.
        // I generally prefer command "execute" methods like this to return `int` which then "bubbles up" to main to exit with
        // `Exit` calls can sometimes make debugging more difficult.
        break;
    }
}
finally
{
    cancelMutex.ReleaseMutex();
}
```

In this case the `try...finally` isn't really necessary; it's more signaling to developers that this section of code is running with the lock.

Reply:
I agree. I'll change the code to that.